simplecnf.go

```diff
@@ -0,0 +1,160 @@
+package go_sat_solver
+
+import "math"
+
+type orClause []VariableAssignment
+
+type simpleSatCnf struct {
+	clauses []orClause
+}
+
+func (f simpleSatCnf) IsSatisfied() bool {
+	return len(f.clauses) == 0
+}
+
+func (f simpleSatCnf) IsContradiction() bool {
+	for _, c := range f.clauses {
+		if len(c) == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+type variableAssignment map[Variable]bool
+
+// putAll adds all the assignments to the map
+func (t variableAssignment) putAll(toAdd map[Variable]bool) {
+	m := map[Variable]bool(t)
+	for k, v := range toAdd {
+		m[k] = v
+	}
+}
+
+func (f simpleSatCnf) UnitPropagate() (CnfFormula, map[Variable]bool, []Variable) {
```

Comment:
seems like a variable assignment (the kind that refers to a variable which has been assigned) should be a publicly visible kind of object, so that this method can take in a variable assignment rather than just a generic map from variables to bools.